ConnectionFactory.php

```diff
@@ -78,7 +77,7 @@ public function createConnection(array $params, ?Configuration $config = null, ?
             }
 
             if (! isset($params['charset'])) {
-                if ($driver instanceof AbstractMySQLDriver) {
+                if ($driver->getDatabasePlatform()->getName() === 'mysql') {
```

Comment:
This is sadly calling a deprecated method but I could not find another method which works on this range of supported doctrine/dbal versions.

Reply:
Maybe by performing several `instanceof` checks?